chia/wallet/trading/offer.py

```diff
@@ -148,6 +150,27 @@ def __post_init__(self) -> None:
             if max_cost < 0:
                 raise ValidationError(Err.BLOCK_COST_EXCEEDS_MAX, "compute_additions for CoinSpend")
         object.__setattr__(self, "_additions", adds)
+        object.__setattr__(self, "_conditions", None)
+
+    def conditions(self) -> List[Condition]:
+        if self._conditions is None:
+            conditions: Dict[Coin, List[Condition]] = {}
+            max_cost = DEFAULT_CONSTANTS.MAX_BLOCK_COST_CLVM
+            for cs in self._bundle.coin_spends:
+                try:
+                    cost, conds = cs.puzzle_reveal.run_with_cost(max_cost, cs.solution)
+                    max_cost -= cost
+                    conditions[cs.coin] = parse_conditions_non_consensus(conds.as_iter())
```

Comment:
at this point, you need to capture the coin's timestamp and creation height and "resolve" the relative time locks based on it.

Maybe we don't have access to the coin database here though.